File: csharp/Onboarding.cs
// https://www.codingame.com/training/easy/onboarding

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Player
{
    static void Main(string[] args)
    {

        // game loop
        while (true)
        {
            string enemy1 = Console.ReadLine(); 
            int dist1 = int.Parse(Console.ReadLine());
            string enemy2 = Console.ReadLine();
            int dist2 = int.Parse(Console.ReadLine());

            Console.WriteLine(dist1 < dist2 ? enemy1 : enemy2);
        }
    }
}

File: go/classic_puzzles_easy/Onboarding.go
// https://www.codingame.com/training/easy/onboarding

package main

import "fmt"

func main() {
	for {
		var enemy1 string
		fmt.Scan(&enemy1)

		var dist1 int
		fmt.Scan(&dist1)

		var enemy2 string
		fmt.Scan(&enemy2)

		var dist2 int
		fmt.Scan(&dist2)

		if dist1 < dist2 {
			fmt.Println(enemy1)
		} else {
			fmt.Println(enemy2)
		}
	}
}
